homeassistant/components/planifneige/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""
+Sensor for the City of Montreal's Planif-Neige snow removal APIs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.planifneige/
+"""
+
+import logging
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.restore_state import RestoreEntity
+
+DEPENDENCIES = ['planifneige']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREETID = 'streetid'
+CONF_STREETS = 'streets'
+CONF_DBPATH = 'database_path'
+
+ATTR_SENSOR = 'sensor'
+ATTR_STATES = 'states'
+
+DOMAIN = 'planifneige'
+
+STREET_STATE = {
+    '0': ['Snowed', 'mdi:snowflake'],  # street is snowed in
+    '1': ['Clear', 'mdi:road'],  # street is clear
+    '2': ['Planned', 'mdi:clock-outline'],  # street is planned for clearing
+    '3': ['Planned', 'mdi:clock-outline'],  # clearing planned with new date
```

Comment:
Differentiating 2 from 3 would be good, something like "Replanned" with `mdi:clock-alert-outline` maybe?

Reply:
This is exactly how I had it set, but then I figured the user doesn't really care that it's getting rescheduled.

Anyhow, I guess it's best to provide all states and let the them decide what to do with it.

Btw, this is how the city describes the states (in french, taken verbatim from [API docs](http://donnees.ville.montreal.qc.ca/dataset/1060866e-1a19-4ded-8273-0e01fec3df6f/resource/34fbca5a-eea6-4fba-9745-188049334fea/download/planifneigeapispecificationsrev19octobre2015.docx)):
```
«0» – Enneigé
«1» – Déneigé
«2» – Planifié
«3» – Replanifié
«4» – Sera replanifié ultérieurement
« 5 » - Chargement en cours
« 10 » - Dégagé (entre 2 chargements de neige)
```
I'm observing the states of the streets around me, when there are snowfalls/storms and trying and understand what 10 really means, and might update the state descriptions in the future.
With some "luck" we'll get enough snowfall before the end of the season.  :)